packages/babel-helper-member-expression-to-functions/src/index.js

```diff
@@ -37,9 +37,113 @@ const handle = {
   handle(member) {
     const { node, parent, parentPath } = member;
 
+    if (member.isOptionalMemberExpression()) {
+      const root = member.find(({ node, parent, parentPath }) => {
+        if (parentPath.isOptionalMemberExpression()) {
+          return parent.optional || parent.object !== node;
+        }
+        if (parentPath.isOptionalCallExpression()) {
+          return parent.optional || parent.callee !== node;
+        }
+        return true;
+      });
+
+      const rootParentPath = root.parentPath;
+      if (
+        rootParentPath.isUpdateExpression({ argument: node }) ||
+        rootParentPath.isAssignmentExpression({ left: node })
+      ) {
+        throw member.buildCodeFrameError(`can't handle assignment`);
+      }
+      if (rootParentPath.isUnaryExpression({ operator: "delete" })) {
+        throw member.buildCodeFrameError(`can't handle delete`);
+      }
+
+      if (node.optional) {
+        throw member.buildCodeFrameError(
+          `can't handle '?.' directly before ${node.property.type}`,
+        );
+      }
+
+      let nearestOptional = member;
+      for (;;) {
+        if (nearestOptional.isOptionalMemberExpression()) {
+          if (nearestOptional.node.optional) break;
+          nearestOptional = nearestOptional.get("object");
+          continue;
+        } else if (nearestOptional.isOptionalCallExpression()) {
+          if (nearestOptional.node.optional) break;
+          nearestOptional = nearestOptional.get("object");
```

Comment:
Shouldn't this be `callee`?

Reply:
Yup, thanks! Looks like I need more tests.